### PR TITLE
[AIRFLOW-1940] Fix boto3 s3 transform operator

### PR DESCRIPTION
**Pull request by UnclaimedPants**
Make sure you have checked _all_ steps below.

### JIRA
- [x ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1940

### Description
- [ x] Here are some details about my PR, including screenshots of any UI changes:

This PR updates boto2 syntax to boto3 for s3 file transform operator

### Tests
- [x ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

2-line fix

### Commits
- [x ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by ashb** [comment]
Please add tests so that this operator doesn't break in the future. For an example to base the tests on check out tests/hooks/test_s3_hook.py and it's used of "moto".

PRs should also target the master branch.

File: airflow/operators/s3_file_transform_operator.py [hunk 1]
```diff
@@ -100,10 +99,9 @@ def execute(self, context):
                 )
             self.log.info("Uploading transformed file to S3")
             f_dest.flush()
-            dest_s3.load_file(
+            dest_s3.upload_file(
```
Comment (ashb) [inline airflow/operators/s3_file_transform_operator.py]: `dest_s3` is an S3Hook object -- the method is called `load_file`, `upload_file` does not exist.

**Comment by Fokko** [comment]
This isn't in contrib, we should have some tests here..

**Comment by artwr** [comment]
Or we should move it to contrib... Maybe in 2.0 ?

File: airflow/operators/s3_file_transform_operator.py (hunk 1 above)
Comment (UnclaimedPants) [reply airflow/operators/s3_file_transform_operator.py]: You're right :) Fixing.

**Comment by Fokko** [comment]
It would be awesome if @UnclaimedPants could provide some simple tests, as a start. To avoid errors such as @ashb pointed out.

**Pull request closed without merging**